src/jdk.attach/macosx/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -67,10 +74,12 @@ public class VirtualMachineImpl extends HotSpotVirtualMachine {
         // Find the socket file. If not found then we attempt to start the
         // attach mechanism in the target VM by sending it a QUIT signal.
         // Then we attempt to find the socket file again.
-        File socket_file = new File(tmpdir, ".java_pid" + pid);
+        // In macOS the socket file is located in per-user temp directory.
+        String tempDir = getTempDirFromPid(pid);
```

Comment:
Nit: I'd suggest to rename it to `tempdir` to follow the local naming convention which prevails.

Reply:
Thanks! Fixed.